dpdispatcher/lsf.py

```diff
@@ -2,141 +2,41 @@
 from abc import ABC
 
 from dpdispatcher.machine import Machine
+from dpdispatcher import dlog
 from dpdispatcher.JobStatus import JobStatus
-from dpdispatcher.submission import Resources
 
 
-lsf_script_template = """\
-{lsf_script_header}
-{lsf_script_env}
-{lsf_script_command}
-{lsf_script_end}
-"""
-
 lsf_script_header_template = """\
 #!/bin/bash -l
 #BSUB -e %J.err
 #BSUB -o %J.out
 {lsf_nodes_line}
 {lsf_ptile_line}
 {lsf_partition_line}
-{lsf_walltime_line}
 {lsf_number_gpu_line}
 """
 
-lsf_script_env_template = """
-{prepend_text}
-export REMOTE_ROOT={remote_root}
-test $? -ne 0 && exit 1
-"""
-
-lsf_script_command_template = """
-cd $REMOTE_ROOT
-cd {task_work_path}
-test $? -ne 0 && exit 1
-if [ ! -f {task_tag_finished} ] ;then
-  {command_env} {command}  1>> {outlog} 2>> {errlog} 
-  if test $? -ne 0; then touch {task_tag_finished}; fi
-  touch {task_tag_finished}
-fi 
-"""
-
-lsf_script_end_template = """
-
-cd $REMOTE_ROOT
-test $? -ne 0 && exit 1
-
-wait
-
-{append_text}
-touch {job_tag_finished}
-"""
-
-lsf_script_wait = """
-wait
-"""
-
-default_lsf_bsub_dict = {
-    'm': "8G"
-}
-
-
-# class LSFResources(Resources):
-#     """
-#     LSF resources
-#     """
-#     def __init__(
-#             self,
-#             number_node,
-#             cpu_per_node,
-#             gpu_per_node,
-#             queue_name,
-#             walltime="120:00:00",
-#             prepend_text="",
-#             append_text="",
-#             gpu_usage=True,
-#             gpu_new_syntax=True,
-#             lsf_bsub_dict=None,
-#             group_size=1
-#     ):
-#         """define LSF resources
-#
-#         Parameters
-#         ----------
-#         number_node: nodes to be used
-#         cpu_per_node: CPU cores used on each node
-#         gpu_per_node: GPU
-#         queue_name: the name of queue
-#         walltime: max time of task
-#         prepend_text: prepend scripts, code executed before the task run
-#         append_text: append scripts, code executed after the task run
-#         gpu_usage: choose if GPU line is used
-#         lsf_bsub_dict: other bsub parameters.
-#         group_size: tasks contained by each group
-#         """
-#         super().__init__(number_node, cpu_per_node, gpu_per_node, queue_name, group_size)
-#         if lsf_bsub_dict is None:
-#             lsf_bsub_dict = {}
-#         self.walltime = walltime
-#         self.gpu_new_syntax = gpu_new_syntax
-#         self.gpu_usage = gpu_usage
-#         self.prepend_text = prepend_text
-#         self.append_text = append_text
-#         self.lsf_bsub_dict = lsf_bsub_dict
 
-
-class LSF(Machine):
+class LSF(Machine, ABC):
```

Comment:
why is it abstract base class?

Reply:
It is a PyCharm issue and I just pressed the button, let PyCharm solve it automatically. `ABC` has been removed here for it is not an abstract base class. And from `Machine`, I found two abstract methods not implemented by the derived class and added.